File: react/frontend/src/components/ContentGenerator.js
import React, { useState } from 'react';
import { generateContent } from '../api/contentApi';
import '../styles/styles.css';

const ContentGenerator = ({ prompt }) => {
  const [tone, setTone] = useState('Neutral');
  const [length, setLength] = useState('Medium');
  const [audience, setAudience] = useState('General');
  const [generatedContent, setGeneratedContent] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(null);

  const toneOptions = ['Neutral', 'Formal', 'Informal', 'Persuasive', 'Enthusiastic'];
  const lengthOptions = ['Short', 'Medium', 'Long'];
  const audienceOptions = ['General', 'Technical', 'Business', 'Creative'];

  const handleToneChange = (e) => setTone(e.target.value);
  const handleLengthChange = (e) => setLength(e.target.value);
  const handleAudienceChange = (e) => setAudience(e.target.value);

  const handleContentGeneration = async () => {
    if (!prompt) {
      setError('Please enter a prompt for content generation.');
      return;
    }

    setIsLoading(true);
    setError(null);

    try {
      const generatedContent = await generateContent(prompt, tone, length, audience);
      setGeneratedContent(generatedContent.generated_content);
    } catch (error) {
      console.error('Error generating content:', error);
      setError('Failed to generate content. Please try again.');
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <div className="component">
      <div>
        <label htmlFor="tone">Tone:</label>
        <select id="tone" value={tone} onChange={handleToneChange}>
          {toneOptions.map((tone) => (
            <option key={tone} value={tone}>{tone}</option>
          ))}
        </select>

        <label htmlFor="length">Length:</label>
        <select id="length" value={length} onChange={handleLengthChange}>
          {lengthOptions.map((length) => (
            <option key={length} value={length}>{length}</option>
          ))}
        </select>

        <label htmlFor="audience">Audience:</label>
        <select id="audience" value={audience} onChange={handleAudienceChange}>
          {audienceOptions.map((audience) => (
            <option key={audience} value={audience}>{audience}</option>
          ))}
        </select>

        <button onClick={handleContentGeneration} disabled={isLoading}>
          {isLoading ? 'Generating...' : 'Generate Content'}
        </button>
      </div>
      {isLoading && <p>Loading...</p>}
      {error && <p>{error}</p>}
      {generatedContent && (
        <div>
          <h3>Generated Content:</h3>
          <p>{generatedContent}</p>
        </div>
      )}
    </div>
  );
};

export default ContentGenerator;

File: react/frontend/src/components/contentGenerator.py
from flask import Flask, render_template, request, jsonify
import requests

app = Flask(__name__)

class ContentGenerator:
    def __init__(self):
        self.prompt = ''
        self.tone = 'Neutral'
        self.generated_content = ''
        self.is_loading = False
        self.error = None

    def generate_content(self):
        try:
            response = requests.post('http://localhost:5003/generate_content', json={'prompt': self.prompt, 'tone': self.tone})
            self.generated_content = response.json()['generated_content']
        except Exception as e:
            print('Content generation error:', e)
            self.error = 'Failed to generate content. Please try again later.'
        finally:
            self.is_loading = False

content_generator = ContentGenerator()

@app.route('/', methods=['GET', 'POST'])
def index():
    if request.method == 'POST':
        content_generator.prompt = request.form['prompt']
        content_generator.tone = request.form['tone']
        content_generator.generate_content()
    return render_template('content_generator.html', content_generator=content_generator)

if __name__ == '__main__':
    app.run(debug=True)
